drivers/clock_control/nrf_power_clock.c

```diff
@@ -577,94 +613,13 @@ DEVICE_AND_API_INIT(clock_nrf, DT_INST_LABEL(0),
 		    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
 		    &clock_control_api);
 
-#if defined(CONFIG_USB_NRFX)
-static bool power_event_check_and_clean(nrf_power_event_t evt, uint32_t intmask)
-{
-	bool ret = nrf_power_event_check(NRF_POWER, evt) &&
-			nrf_power_int_enable_check(NRF_POWER, intmask);
-
-	if (ret) {
-		nrf_power_event_clear(NRF_POWER, evt);
-	}
-
-	return ret;
-}
-#endif
-
-static void usb_power_isr(void)
-{
-#if defined(CONFIG_USB_NRFX)
-	extern void usb_dc_nrfx_power_event_callback(nrf_power_event_t event);
-
-	if (power_event_check_and_clean(NRF_POWER_EVENT_USBDETECTED,
-					NRF_POWER_INT_USBDETECTED_MASK)) {
-		usb_dc_nrfx_power_event_callback(NRF_POWER_EVENT_USBDETECTED);
-	}
-
-	if (power_event_check_and_clean(NRF_POWER_EVENT_USBPWRRDY,
-					NRF_POWER_INT_USBPWRRDY_MASK)) {
-		usb_dc_nrfx_power_event_callback(NRF_POWER_EVENT_USBPWRRDY);
-	}
-
-	if (power_event_check_and_clean(NRF_POWER_EVENT_USBREMOVED,
-					NRF_POWER_INT_USBREMOVED_MASK)) {
-		usb_dc_nrfx_power_event_callback(NRF_POWER_EVENT_USBREMOVED);
-	}
-#endif
-}
-
-void nrf_power_clock_isr(void *arg)
-{
-	ARG_UNUSED(arg);
-	struct device *dev = DEVICE_GET(clock_nrf);
-
-	if (clock_event_check_and_clean(NRF_CLOCK_EVENT_HFCLKSTARTED,
-					NRF_CLOCK_INT_HF_STARTED_MASK)) {
-		struct nrf_clock_control_sub_data *data =
-				get_sub_data(dev, CLOCK_CONTROL_NRF_TYPE_HFCLK);
-
-		/* Check needed due to anomaly 201:
-		 * HFCLKSTARTED may be generated twice.
-		 *
-		 * Also software should be notified about clock being on only
-		 * if generic request occured.
-		 */
-		if ((GET_STATUS(data->flags) == CLOCK_CONTROL_STATUS_STARTING)
-			&& (hfclk_users & HF_USER_GENERIC)) {
-			clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_HFCLK);
-		}
-	}
-
-	if (clock_event_check_and_clean(NRF_CLOCK_EVENT_LFCLKSTARTED,
-					NRF_CLOCK_INT_LF_STARTED_MASK)) {
-		if (IS_ENABLED(
-			CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION)) {
-			z_nrf_clock_calibration_lfclk_started();
-		}
-		clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_LFCLK);
-	}
-
-	usb_power_isr();
-
-	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION)) {
-		z_nrf_clock_calibration_isr();
-	}
-}
-
 #ifdef CONFIG_USB_NRFX
 void nrf5_power_usb_power_int_enable(bool enable)
 {
-	uint32_t mask;
-
-	mask = NRF_POWER_INT_USBDETECTED_MASK |
-	       NRF_POWER_INT_USBREMOVED_MASK |
-	       NRF_POWER_INT_USBPWRRDY_MASK;
-
 	if (enable) {
-		nrf_power_int_enable(NRF_POWER, mask);
-		irq_enable(DT_INST_IRQN(0));
+		nrfx_power_usbevt_enable();
 	} else {
-		nrf_power_int_disable(NRF_POWER, mask);
+		nrfx_power_usbevt_disable();
```

Comment:
Why not entirely remove `nrf5_power_usb_power_int_enable()` (which is quite an artificial thing here) and call those functions directly from `usb_dc_nrfx.c`?

Reply:
moved